why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,185 @@
+released in 2024, although the first version was released in 2019. The
+developers have set it up as a not-for-profit foundation in the
+Netherlands.
+
+::: columns
+::: column
+#### Benefits
+
+-   Although a very young project it is already in use by a number of
+    large companies and has a large community of users.
+    -   The code is open source, and the way it has been set up is
+        promising for it to remain that way.
+-   It is very fast, and can handle large data sets.
+:::
+
+::: column
+#### Drawbacks
+
+-   It is not designed to be used as a data cleaning tool, but as a
+    database.
+:::
+:::
+
+### Polars
+
+The Polars project started in 2020 and is based in the Netherlands.
+Relatively small team but seems to have a fast growing user base, and
+very enthusiastic developers. Polars is written in Rust and available
+for Python. It is designed to be faster than Pandas, and to handle all
+common data formats. The two main developers are described as
+enthusiastic about open source, which makes it a potentially good choice
+for us.
+
+::: columns
+::: column
+#### Benefits
+
+-   Enthusiastic user community, with a lot of support available.
+-   Supports several languages, including Python.
+-   Handles all common data formats.
+-   Faster than Pandas.
+:::
+
+::: column
+#### Drawbacks
+
+-   Relatively young project, so may not have all the features of more
+    established tools.
+-   Not as well known as Pandas, but certainly getting there.
+-   Compared with Pandas there aren't as many online courses available.
+:::
+:::
+
+### Pandas
+
+Pandas was initially released in 2008, and is now a well established
+tool to work with data frames in Python. The tool is well supported, and
+has a large user base. It is also well documented, and there are a lot
+of online courses available for users to learn how to use it. The tool
+is written in Python, and is designed to be easy to use, and to work
+with large data sets. The main critique is that it can be slow when
+working with large data sets.
+
+::: columns
+::: column
+#### Benefits
+
+-   Established tool with a large user base.
+-   Well supported, and well documented.
+-   Works with all relevant languages and data formats.
+:::
+
+::: column
+#### Drawbacks
+
+-   Can be slow when working with large data sets.
+-   May see itself overtaken by newer tools in future.
+:::
+:::
+
+## Decision outcome
+
+::: content-hidden
+What decision was made, use the form "We decided on CHOICE because of
+REASONS."
+:::
+
+We have decided to use Polars for data cleaning, as it is a fast and well documented, and we expect that we may use it elsewhere in the project. It is also a relatively new tool, but it is already well supported with an enthusiastic following, which we expect will continue in future. Our second choice would be Pandas, not because DuckDB isn't a good tool (we may use it elsewhere in the project), but because it is not designed to be used as a data cleaning tool.
```

Comment:
```suggestion
We have decided to use Polars for data cleaning, as it is a fast and well documented, and we expect that we may use it elsewhere in the project. It is also a relatively new tool, but it is already well supported with an enthusiastic following, which we expect will continue in future.
```

This second part of the paragraph isn't necessary, nor completely true (see comment above about DuckDB and data cleaning).